### ISSUE TITLE
comment is visible when some feature like pick or length

### ISSUE THREAD

**Issue by gabs77** [mikefarah/yq#1231]
Comment is visible when some feature like pick or length

Version of yq: 4.25.X
Operating system: linux/windows

**Input Yaml**
data1.yml:
```yaml
# --------------------------------------------------
# It's a test with comment
# --------------------------------------------------
test:
  - name: a
  - name: b
  - name: c
groups:
  - name: a
  - name: b
  - name: c
  - name: c
```
data2.yml:
```yaml
# --------------------------------------------------
# It's a test with comment
# --------------------------------------------------
groups:
  - name: d
```
**Command**
The command you ran:
```
 yq '[ .groups[].name | select(.=="a") ] | length' data1.yml
```

**Actual behavior**

```yaml
# --------------------------------------------------
# It's a test with comment
# --------------------------------------------------
1
```

**Expected behavior**

```yaml
1
```

**Additional context**
Same problem with the pick feature
yq 'pick(["groups"]) | explode(.) as $item ireduce ({}; . *+ $item)' data1.yml data2.yml --no-doc --prettyPrint

**Actual behavior**

```yaml
# --------------------------------------------------
# It's a test with comment
# --------------------------------------------------
groups:
  - name: a
  - name: b
  - name: c
  - name: c
# --------------------------------------------------
# It's a test with comment
# --------------------------------------------------
groups:
  - name: d
```

**Expected behavior**

```yaml
groups:
  - name: a
  - name: b
  - name: c
  - name: c
groups:
  - name: d
```

**Comment by mikefarah** [comment]
Hi - I agree with the length operator, the comments should not come through will fix in the next release. Till then you can
'[ .groups[].name | select(.=="a") ] | length | . headComment = ""'

For `pick` I actually think keeping the comments makes sense - as it's basically a filter on the map/array (rather than a new calculated result). Use `. headComment = ""` or even `... comments=""` to remove the comments.

**Comment by gabs77** [comment]
Perfect
Thanks

**Comment by oramirite** [comment]
Thanks, this crept up on me today and I thought I was going crazy. Seems to affect has() as well. Got really confusing, since it should really just be a sanitized true/false no matter what.

**Comment by mikefarah** [comment]
Fixed in v4.25.3
